pkg/devices/host.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package devices
+
+import (
+	"github.com/jaypipes/ghw"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+)
+
+type hostDevice struct {
+	types.APIDevice
+	basePciDevice *ghw.PCIDevice
+	driver        string
+}
+
+// NewHostDevice returns an instance of HostDevice interface
+// A list of DeviceInfoProviders can be set externally.
+// If empty, the default driver-based selection provided by ResourceFactory will be used
+func NewHostDevice(dev *ghw.PCIDevice, deviceID string, rFactory types.ResourceFactory,
+	rc *types.ResourceConfig, infoProviders []types.DeviceInfoProvider) (types.HostDevice, error) {
+	// Get driver info
+	driverName, err := utils.GetDriverName(dev.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use the default Information Provided if not
+	if len(infoProviders) == 0 {
+		infoProviders = append(infoProviders, rFactory.GetDefaultInfoProvider(deviceID, driverName))
+	}
+
+	apiDevice := NewAPIDevice(deviceID, infoProviders)
+
+	if !rc.ExcludeTopology {
+		nodeNum := utils.GetDevNode(dev.Address)
+		if nodeNum >= 0 {
+			topology := &pluginapi.TopologyInfo{
+				Nodes: []*pluginapi.NUMANode{
+					{ID: int64(nodeNum)},
+				},
+			}
+			apiDevice.GetAPIDevice().Topology = topology
+		}
+	}
+
+	return &hostDevice{
+		APIDevice:     apiDevice,
+		basePciDevice: dev,
```

Comment:
I am trying to understand why you need the basePCIDevice in the hostDevice.
if I understand the refactor the pci is save in the GenPCIDevice right?
Are you keeping it to get the vendor and product id?

Reply:
Yes, it is for vendor and product ID.

Reply:
so why NewHostDevice need the full ghw.PCIDevice? You can pass just the vendor and Product ID